orchestration/src/test/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatModelTest.java

```diff
@@ -133,4 +141,50 @@ void testStreamCompletion() throws IOException {
       Mockito.verify(inputStream, times(1)).close();
     }
   }
+
+  @Test
+  void testToolCalls() throws IOException {
+    // https://platform.openai.com/docs/guides/function-calling
+    stubFor(
+        post(urlPathEqualTo("/completion"))
+            .inScenario("Tool Calls")
+            .whenScenarioStateIs(STARTED)
+            .willReturn(
+                aResponse()
+                    .withBodyFile("toolCallsResponse.json")
+                    .withHeader("Content-Type", "application/json"))
+            .willSetStateTo("Second Call"));
+
+    stubFor(
+        post(urlPathEqualTo("/completion"))
+            .inScenario("Tool Calls")
+            .whenScenarioStateIs("Second Call")
+            .willReturn(
```

Comment:
(Minor)

There's an easier way to chain subsequent request stubbing, e.g. chaining `.willReturn(..)`

Reply:
chaining only takes the second willReturn value, the first one is forgotten. Chaining only works with Mockito